feat: implement user-defined command aliases via config file

thatonecodes opened the pull request:
This PR implements user-defined command aliases using the `~/.ggcconfig` file.

## Summary
- Added support for defining aliases in ~/.ggcconfig.yaml
- Aliases can map to a single command (e.g., `st: status`) or a sequence of commands (e.g., `acp: ["add", "commit", "push"]`)
- Implemented alias parsing and execution in router
- Commands from aliases run sequentially and stop on failure
- Updated config validation and tests to support mixed alias types
- Extended `config` command to display alias configuration in readable format

## Related Issue
Closes #64.

## Checklist
- [x] I have read the [CONTRIBUTING.md](https://github.com/ggc-dev/ggc/blob/main/CONTRIBUTING.md)
- [x] I have added or updated tests
- [x] I have updated the documentation (if required)
- [x] Code is formatted with `make fmt`
- [x] Code passes linter checks via `make lint`
- [x] All tests are passing

## Additional Context
Aliases are able to be run depending on user's configuration file in `~/.ggcconfig.yaml`. As an example:
```yaml
aliases:
    ac:
        - add .
        - commit tmp
    br: branch
    ci: commit
    quick:
        - status
        - add .
        - commit tmp
    st: status
    sync:
        - pull current
        - add .
        - commit tmp
        - push current
```
This is the default config and has `ggc ac`, `ggc ci` and others, which will execute commands sequentially in order based on the list.

bmf-san commented on config/config.go:
```diff
@@ -216,12 +347,15 @@ func getDefaultConfig() *Config {
 	config.Aliases["br"] = "branch"
 	config.Aliases["ci"] = "commit"
 
+	config.Aliases["ac"] = []interface{}{"add .", "commit tmp"}
```
I noticed a bug while reviewing this implementation, so I created an issue. 🚀 
https://github.com/bmf-san/ggc/issues/120

bmf-san commented on config/config.go:
```diff
@@ -216,12 +347,15 @@ func getDefaultConfig() *Config {
 	config.Behavior.StashBeforeSwitch = true
 
-	// Default aliases
+	// Default simple aliases
```
While reviewing the implementation, I noticed that we're currently setting several default values in the configuration—particularly aliases—that users might not need or expect. I'd like to propose a small adjustment to our default configuration strategy.

### Current concern

The implementation provides a number of default aliases (`st`, `br`, `ci`, `ac`, `sync`, `quick`) that are automatically included, even when users define their own configuration files. This could lead to:

* Unexpected behavior for users unaware of these defaults
* Reduced user control over their own setup
* A departure from the principle of least surprise

### Proposed approach

We should consider limiting default values to only those necessary for the tool’s core functionality. For instance:

* `Default.Branch = "main"` (for repository initialization)
* `Default.Editor = "vim"` (for editing commit messages)
* `Default.MergeTool = "vimdiff"` (for resolving merge conflicts)

Optional settings—such as aliases—should be left empty by default, allowing users to define them explicitly as needed.

### Benefits

* Greater user control over configuration
* Fewer surprises from implicit defaults
* A cleaner, more predictable user experience
* Alignment with the principle of providing only essential defaults

Would you be open to updating the default configuration to include only the minimum necessary settings?
I think this change would improve clarity and flexibility for end users. Let me know your thoughts.

bmf-san commented on config/config.go:
```diff
@@ -140,10 +156,66 @@ func (c *Config) validateIntegrationTokens() error {
 
 func (c *Config) validateAliases() error {
-	for name := range c.Aliases {
+	validCommands := map[string]bool{
```
Isn't the validCommand function actually unnecessary? 👀 

The existence of the command is already checked on the router.go side, so it seemed unnecessary here.

thatonecodes commented:
As stated in the code review, `validCommand` was unneeded, so I removed it, and I also removed the default aliases to prevent the saving/updating bug and to give more clarity and flexibility for end users. I also cleaned up any merge conflicts with the main branch. Please let me know if there are any more issues 👍

bmf-san reviewed (approved):
@thatonecodes Thanks for making the adjustments based on the review comments. LGTM! :D
